massdriver-application-azure-function-app/main.tf

```diff
@@ -4,12 +4,20 @@ locals {
 }
 
 module "application" {
-  source                  = "github.com/massdriver-cloud/terraform-modules//massdriver-application?ref=22d422e"
+  source                  = "github.com/massdriver-cloud/terraform-modules//massdriver-application?ref=4001e6c"
   name                    = var.md_metadata.name_prefix
   service                 = "function"
   application_identity_id = azurerm_linux_function_app.main.identity[0].principal_id
   # We aren't creating an application identity for this module because we are assigning permissions directly to the system-assigned managed identity of the function app.
   create_application_identity = false
+  # The permission-assignment goes like this
+  # Azure makes the function
+  # MDXC tries to assign the role
+  # The storage account is not ready yet
+  # Without this depends_on we get intermitent, hard to debug failures.
+  depends_on = [
+    azurerm_storage_account.main
```

Comment:
🍌 s fix holy cow. I've seen this a few times but was finally able to narrow it down.

```
│ Error: loading Role Definition List: authorization.RoleDefinitionsClient#List: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceNotFound" Message="The Resource 'Microsoft.Storage/storageAccounts/localdevsto000' under resource group 'local-dev-sto-000' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
│
│   with module.application.module.application.mdxc_application_permission.main[".connections.storage.data.security.iam.\"read/write\""],
│   on .terraform/modules/application.application/massdriver-application/main.tf line 64, in resource "mdxc_application_permission" "main":
│   64: resource "mdxc_application_permission" "main" {
│
```

I've learned to not trust implicit dependencies from the `azurerm` provider and I didn't even know that was possible, is that possible?